add pca9685 parameter again

christianrauch opened the pull request:
### Solved Problem
Fixes https://github.com/PX4/PX4-Autopilot/issues/21929 for `beta` branch.

Fixes #21929

### Solution
Add parameter `PCA9685_RATE` again to set `_targetFreq`.

### Changelog Entry
For release notes:
```
Feature/Bugfix add parameter PCA9685_RATE as a replacement for removed PWM_MAIN_RATE again
New parameter: PCA9685_RATE
```

### Test coverage
Tested with PCA9685 breakout board.

dagar commented:
The parameter (PWM_MAIN_RATE) is named way to broadly, can you please make it `PCA9685_` specific?

Then first do a pull request to the main branch and once that settles we can cherry-pick it back for release/1.14.

christianrauch commented:
@dagar I renamed the parameter and rebased it onto the main branch. Note: Parameter names can have a maximum of 16 characters. The remaining build error does not look related to this PR.
